odoo/addons/base/models/res_users.py

```diff
@@ -210,8 +210,8 @@ class Users(models.Model):
     SELF_READABLE_FIELDS = ['signature', 'company_id', 'login', 'email', 'name', 'image_1920', 'image_1024', 'image_512', 'image_256', 'image_128', 'lang', 'tz', 'tz_offset', 'groups_id', 'partner_id', '__last_update', 'action_id']
 
     def _default_groups(self):
-        default_user = self.env.ref('base.default_user', raise_if_not_found=False)
-        return (default_user or self.env['res.users']).sudo().groups_id
+        default_user_id = self.env['ir.model.data'].xmlid_to_res_id('base.default_user', raise_if_not_found=False)
+        return self.env['res.users'].browse(default_user_id).sudo().groups_id
```

Comment:
we are relying on the fact that `self.browse(None).groups_id` returns an empty recordset, as you are modifying this line, we can take the opportunity not to rely on ORM side effects
```suggestion
        return default_user_id and self.env['res.users'].browse(default_user_id).sudo().groups_id or []
```

Reply:
isn't it an intended ORM behavior, why can't we just rely on it instead of putting fallbacks everywhere ?

Reply:
Intended ORM behaviour? The fact that `browse` ignores falsy id (but not falsy id**s**) looks more like an side effect of the implementation. I am not sure I would rely on it to save bits of code. Anyway, just a detail, I won't _chier une pendule_ if you disagree.